python/nano/src/bigdl/nano/automl/hpo/backend.py

```diff
@@ -83,22 +83,28 @@ def get_hpo_config(trial, configspace):
             hp_prefix = hp_obj.meta.setdefault('prefix', None)
             hp_name = hp_prefix + ':' + hp if hp_prefix else hp
             hp_type = str(type(hp_obj)).lower()  # type of hyperparam
-            if 'integer' in hp_type:
-                hp_dimension = trial.suggest_int(
-                    name=hp_name, low=int(hp_obj.lower), high=int(hp_obj.upper))
-            elif 'float' in hp_type:
-                if hp_obj.log:  # log10-scale hyperparmeter
-                    hp_dimension = trial.suggest_loguniform(
-                        name=hp_name, low=float(hp_obj.lower), high=float(hp_obj.upper))
-                else:
-                    hp_dimension = trial.suggest_float(
-                        name=hp_name, low=float(hp_obj.lower), high=float(hp_obj.upper))
-            elif 'categorical' in hp_type:
-                hp_dimension = trial.suggest_categorical(
-                    name=hp_name, choices=hp_obj.choices)
-            elif 'ordinal' in hp_type:
-                hp_dimension = trial.suggest_categorical(
-                    name=hp_name, choices=hp_obj.sequence)
+            if 'integer' in hp_type or 'float' in hp_type or \
+                    'categorical' in hp_type or 'ordinal' in hp_type:
+                try:
+                    if 'integer' in hp_type:
+                        hp_dimension = trial.suggest_int(
+                            name=hp_name, low=int(hp_obj.lower), high=int(hp_obj.upper))
+                    elif 'float' in hp_type:
+                        if hp_obj.log:  # log10-scale hyperparmeter
+                            hp_dimension = trial.suggest_loguniform(
+                                name=hp_name, low=float(hp_obj.lower), high=float(hp_obj.upper))
+                        else:
+                            hp_dimension = trial.suggest_float(
+                                name=hp_name, low=float(hp_obj.lower), high=float(hp_obj.upper))
+                    elif 'categorical' in hp_type:
+                        hp_dimension = trial.suggest_categorical(
+                            name=hp_name, choices=hp_obj.choices)
+                    elif 'ordinal' in hp_type:
+                        hp_dimension = trial.suggest_categorical(
+                            name=hp_name, choices=hp_obj.sequence)
+                except (ValueError):
+                    raise ValueError("You have set searching space  \
```

Comment:
Change the error message to: "If you set search space in model, you must call model.search before model.fit, "